### DISCUSSION TITLE
Why is MQTT server connection required?

### DISCUSSION THREAD

**Discussion by Littlesloths** [jziolkowski/tdm#262] [General]
Hello :-)
I am wondering why is a MQTT connection required? Most of my devices do not write back data to a MQTT server, but it would be interesting to keep track of Tasmota versions, rules, etc.
It looks like I need to send values from the device to MQTT server + store the signals?

My suggestion would be to scan a given IP range and do the same evaluation without MQTT. Does that make sense?

**Comment by jziolkowski** [comment]
Because "m" in tasmota stands for mqtt. It's the primary method of connecting with tasmotized devices. There are plenty of advantages of mqtt over http, which has been outlined many times.

With that said I did consider detecting tasmota devices based on IP communication, however: repeatedly querying them over http puts much more strain on the devices as compared to mqtt. Also the whole application logic of tdm is centered around mqtt messages.

If you insist on http access, you can give tasmoadmin a try, but be aware that it's been long known that it makes the devices unstable quite often.

**Reply by Littlesloths** [reply]
Thank you for the quick response. Sorry for having raised that question, but I could not find previous related discussions...

So, the minimum requirement to see the devices is:
a) send values from device to MQTT server
b) store the values to e.g. influxdb?

**Reply by jziolkowski** [reply]
Don't be sorry, it's a valid question. I'm happy to explain.

You don't need influxdb unless you want to store reported values for a longer time. For normal operation mqtt is more than enough, granted (by design of mqtt) you will only see the values at the time when you are observing them. Ergo tdm doesn't retrieve any historical data (or store any, for that matter); it only shows what tasmota devices report in their stays messages, telemetry and generic responses.
